src/plugins/output_format/common.h

```diff
@@ -0,0 +1,241 @@
+* reuse, modify, and relicense the code.  DRAKVUF will always be          *
+* available Open Source, but this is important because the inability to   *
+* relicense code has caused devastating problems for other Free Software  *
+* projects (such as KDE and NASM).                                        *
+* To specify special license conditions of your contributions, just say   *
+* so when you send them.                                                  *
+*                                                                         *
+* This program is distributed in the hope that it will be useful, but     *
+* WITHOUT ANY WARRANTY; without even the implied warranty of              *
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+* license file for more details (it's in a COPYING file included with     *
+* DRAKVUF, and also available from                                        *
+* https://github.com/tklengyel/drakvuf/COPYING)                           *
+*                                                                         *
+***************************************************************************/
+
+#ifndef PLUGINS_OUTPUT_FORMAT_COMMON_H
+#define PLUGINS_OUTPUT_FORMAT_COMMON_H
+
+#include "ostream.h"
+
+#include <libdrakvuf/libdrakvuf.h>
+#include <plugins/private.h>
+
+#include <algorithm>
+#include <functional>
+#include <iomanip>
+#include <ios>
+#include <iostream>
+#include <map>
+#include <optional>
+#include <sstream>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+struct TimeVal
+{
+    glong tv_sec;
+    glong tv_usec;
+};
+
+template<class Value>
+auto keyval(const char* key, const Value& value)
+{
+    return std::make_pair(key, value);
+}
+
+namespace fmt
+{
+
+template<class T>
+struct __val_holder
+{
+    T value;
+
+    __val_holder(T&& v): value(std::forward<T>(v)) {}
+};
+
+/* numeric value */
+template<class T>
+struct Nval: __val_holder<T>
+{
+    Nval(T&& v): __val_holder<T>(std::forward<T>(v)) {}
+};
+
+template<class T, std::enable_if_t<std::is_integral_v<std::remove_reference_t<T>>, int> = 0>
+auto nval(T&& value)
+{
+    return Nval<T>(std::forward<T>(value));
+}
+
+/* format specific numeric value */
+template<class T>
+struct Xval: __val_holder<T>
+{
+    bool withbase;
+    Xval(T&& v, bool use_base = true): __val_holder<T>(std::forward<T>(v)), withbase(use_base) {}
+};
+
+template<class T, std::enable_if_t<std::is_integral_v<std::remove_reference_t<T>>, int> = 0>
+auto xval(T&& value, bool use_base = true)
+{
+    return Xval<T>(std::forward<T>(value));
+}
+
+/* floating value in fixed format */
+template<class T>
+struct Fval: __val_holder<T>
+{
+    Fval(T&& v): __val_holder<T>(std::forward<T>(v)) {}
+};
+
+template<class T, std::enable_if_t<std::is_floating_point_v<T>, int> = 0>
+auto fval(T&& value)
+{
+    return Fval<T>(std::forward<T>(value));
+}
+
+/* raw string value */
+template<class T>
+struct Rstr: __val_holder<T>
```

Comment:
Why have "raw string" and "quoted strings"? This could lead to problems in parsing output files (due to unexpected/missing quotes). I'd propose just "string" (and let the format printer choose, for example: CSV without quotes, JSON with quotes, etc.)

Reply:
KV format can print string with and without quotes. I've tried to support backward compatibility.
It can be removed later by smaller pieces, so wont be so painfully.